app/controllers/volunteer_ops_controller.rb

```diff
@@ -78,27 +78,34 @@ def authorize
     # set @organisation
     # then can make condition:
     # unless current_user.can_edit? organisation
-    unless org_owner? || superadmin?
+
+    unless org_owner?
       flash[:error] = 'You must be signed in as an organisation owner or ' \
                       'site superadmin to perform this action!'
       (redirect_to '/') && return
     end
   end
 
   def org_owner?
-    if params[:organisation_id].present? && current_user_has_organisation?
-      current_user.organisation.friendly_id == params[:organisation_id]
-    elsif current_user_has_organisation?
-      current_user.organisation == VolunteerOp.find(params[:id]).organisation
-    end
-  end
 
-  def current_user_has_organisation?
-    current_user.present? && current_user.organisation.present?
-  end
+    current_user.present? && (current_user.can_edit? set_organisation)
 
+  end
+  
+  # def find_org
+  #   if params[:organisation_id].present?
+  #     organisation = set_organisation
+  #   else
+  #     organisation = VolunteerOp.find(params[:id]).organisation
+  #   end
+  # end
+  
   def set_organisation
```

Comment:
I just have a slight concern over this, since set_organisation is used in a few methods here, and I think we're adding some extra functionality - although tests are passing, so probably okay, but do you know what I mean - there's the two filters - `set_organisation` and `authorise` which have different hooks to the various methods ...